iot-hub/Samples/service/CleanUpDevicesSample/CleanUpDevicesSample.csproj

```diff
@@ -1,12 +1,13 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
     <TargetFramework>netcoreapp2.1</TargetFramework>
+    <LangVersion>8</LangVersion>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.Devices" Version="1.*" />
+    <PackageReference Include="Microsoft.Azure.Devices" Version="1.20.0" />

```

Comment:
How could they possibly be better? They are indeterministic. This is not a pattern for success. No one should do this in their prod app, right?

Reply:
For a production app, sure; you would want to take a dependency on a particular package. 
For a sample demonstrating code usage, if we take a dependency on a particular version, we would have to keep updating the sample any time that we release a new nuget package. Floating version there would ensure that the latest package gets picked up from the feed.

Reply:
What I am focusing here is that sample is demonstrating code usage, and not nuget package usage/ project configuration. I don't want to put the wrong sample out there just because it makes our work easier; but what I am saying is that the customer is looking at our code and not csproj configuration.

Reply:
Right, and I want our samples to follow these production practices.

We don't have to update to the latest package version every time we ship. I suggest we update to a latest version any time we make use of any new API surface. Otherwise the referenced release is still good and users can use it.

I think our release version advances should be explicit and chosen, not automatically selected without testing validation.